src/Form/molecules/FilterDropdown.tsx

```diff
@@ -0,0 +1,537 @@
+    newList.push(selected);
+    return newList;
+  }
+
+
+  if (Array.isArray(selected)) {
+
+    if (selected.length > maxItems) {
+      return selected.slice(0, maxItems - 1);
+    }
+
+    if (selected.length === maxItems) {
+      return selected;
+    }
+    const selectedIndexList: number[] = [];
+
+    selected.forEach((element: IFilterItem) => {
+      const index = list.findIndex(item => {
+        if (isListItem(item) && isListItem(element)) {
+          item.value === element.value;
+        } else {
+          return item === element;
+        }
+      })
+
+      if (index !== -1) {
+        selectedIndexList.push(index);
+      }
+    })
+    selectedIndexList.sort(function (a, b) {
+      return a - b;
+    });
+
+    let selectedIndex = 0;
+    let visibleListAvailability = maxItems - selected.length;
+    const newList: IFilterItem[] = [];
+
+    for (let listIndex = 0; listIndex < list.length; listIndex++) {
+      if ((selectedIndex < selectedIndexList.length) && (listIndex === selectedIndexList[selectedIndex])) {
+        selectedIndex++;
+      } else {
+        visibleListAvailability--;
+      }
+      newList.push(list[listIndex]);
+
+      if (visibleListAvailability === 0) {
+        break;
+      }
+    }
+
+    if ((newList.length < maxItems) && (selectedIndex < selectedIndexList.length)) {
+      for (; selectedIndex < selectedIndexList.length; selectedIndex++) {
+        newList.push(list[selectedIndexList[selectedIndex]]);
+      }
+    }
+
+    return newList;
+  }
+
+  return list.slice(0, maxItems - 1);
+}
+
+const getFilteredList = (list: IFilterItem[], newValue: string): IFilterItem[] => {
+  return list.filter(element => {
+
+    if (isListItem(element) && typeof element.value === 'number') {
+      const valueString = (element.value).toString(10);
+      return valueString.includes(newValue);
+    }
+
+    if (isListItem(element) && typeof element.value === 'string') {
+      return element.value.includes(newValue);
+    }
+
+    if (typeof element === 'number') {
+      const valueString = element.toString(10);
+      return valueString.includes(newValue);
+    }
+
+    if (typeof element === 'string') {
+      return element.includes(newValue);
+    }
+
+  });
+}
+
+const getResultText = (template: string, visible: number, total: number) => {
+  const newMessage = template.replace('[TOTAL]', `${total}`);
+  return newMessage.replace('[VISIBLE]', `${visible}`);
+}
+
+type IFilterItem = string | ListItem | number;
+
+export type IFilterDropdownValue = string | string[] | number | number[] | ListItem | ListItem[] | null;
+
+interface ListItem {
+  text: string;
+  value: string | number;
+}
+
+type IOpenPos = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right'
+
+interface IDropOpen {
+  isOpen: boolean,
+  position: IOpenPos,
+}
+interface IFilterDropdown {
+  buttonIcon: string
+  buttonText: string
+  list: IFilterItem[];
+  selected: IFilterDropdownValue;
+  buttonSize?: TypeButtonSizes
```

Comment:
Shouldn't be needed right now (and not plans for it, unless I missed some thinking  here)